target/arc/semfunc-v2.c

```diff
@@ -8669,114 +8668,136 @@ arc_gen_VMAC2HU(DisasCtxt *ctx, TCGv dest, TCGv b, TCGv c)
  * VADD: VADD2, VADD2H, VADD4H
  */
 
+#define ARC_GEN_BASE_I64(NAME, OP, SIZE)                                            \
+static void                                                                         \
+arc_gen_##NAME##_base_i64(DisasCtxt *ctx, TCGv_i64 dest, TCGv_i64 b, TCGv_i64 c,    \
+                       TCGv_i64 acc, bool set_n_flag,                               \
+                       ARC_GEN_EXTRACT_BITS_FUNC extract_bits,                      \
+                       ARC_GEN_OVERFLOW_DETECT_FUNC detect_overflow_i64)            \
+{                                                                                   \
+    ARC_GEN_VEC_FIRST_OPERAND(operand_##SIZE, i64, b);                              \
+    ARC_GEN_VEC_SECOND_OPERAND(operand_##SIZE, i64, c);                             \
+                                                                                    \
+    OP(dest, b, c);                                                                 \
+}
+
+ARC_GEN_BASE_I64(vadd2, tcg_gen_vec_add32_i64,    32bit);
+ARC_GEN_BASE_I64(vadd4h, tcg_gen_vec_add16_i64,   16bit);
+ARC_GEN_BASE_I64(vsub2, tcg_gen_vec_sub32_i64,    32bit);
+ARC_GEN_BASE_I64(vsub4h, tcg_gen_vec_sub16_i64,   16bit);
+
+
+
+ARC_GEN_32BIT_INTERFACE(VADD2, PAIR, PAIR, PAIR, UNSIGNED, \
+                        arc_gen_vadd2_base_i64);
+
 int
 arc_gen_VADD2(DisasCtxt *ctx, TCGv dest, TCGv b, TCGv c)
 {
-  TCGv cc_temp = tcg_temp_local_new();
-  TCGLabel *cc_done = gen_new_label();
+    ARC_GEN_SEMFUNC_INIT();
 
-  getCCFlag(cc_temp);
-  tcg_gen_brcondi_tl(TCG_COND_EQ, cc_temp, 0, cc_done);
+    arc_autogen_base32_VADD2(ctx, dest, b, c);
 
-  arc_gen_vec_pair_i32(ctx, dest, b, c,
-                       tcg_gen_vec_add32_i64);
+    ARC_GEN_SEMFUNC_DEINIT();
 
-  gen_set_label(cc_done);
-  tcg_temp_free(cc_temp);
+    return DISAS_NEXT;
```

Comment:
It looks like you have some issues with indentation. Please run `scripts/checkpatch.pl` for conformity. Also, please check QEMU official page on coding style: https://www.qemu.org/docs/master/devel/style.html#coding-style